util/settings/settings.go

```diff
@@ -848,19 +848,19 @@ func updateSettingsFromSecret(settings *ArgoCDSettings, argoCDSecret *apiv1.Secr
 		errs = append(errs, &incompleteSettingsError{message: "server.secretkey is missing"})
 	}
 	if githubWebhookSecret := argoCDSecret.Data[settingsWebhookGitHubSecretKey]; len(githubWebhookSecret) > 0 {
-		settings.WebhookGitHubSecret = string(githubWebhookSecret)
+		settings.WebhookGitHubSecret = strings.TrimSpace(string(githubWebhookSecret))
```

Comment:
I'm unsure whether we should mutate the values stored in `settings` here and write them back to the `settings` struct. The `argocd-secret` will be updated with these settings, and this could lead to Out-of-Sync conditions when Argo CD is managing itself in Git (see https://github.com/argoproj/argo-cd/blob/d4ef7444545549cd4763bc8bca3f26f1899d946e/util/settings/settings.go#L921)

On the other hand, this could be used as an indicator that the stored password has unwanted trailing CRLF. But we should have this documented, at least in the FAQ (for example, "Why is my argocd-secret out of sync?")

Reply:
Thank you @jannfis for the prompt review, I personally think this might be confusing to users who have a well set up argo-cd. I think just logging the warning about the extra CRLF characters in the secret string might suffice.